src/taskpane/server.js

```diff
@@ -1128,14 +1220,27 @@ function contentFormattingSetter(sheet, model) {
 
     // protect column
     // read only fields - ie ones you can get from Spira but not create in Spira (as with IDs - eg task component)
-    if (model.fields[i].unsupported || model.fields[i].isReadOnly) {
+    if ((model.fields[i].unsupported || model.fields[i].isReadOnly) && !model.fields[i].isHidden) {

```

Comment:
I see how your code is working but it is a bit confusing. You should split the protectColumn function into two - each does a specific function and only one
1. protectColumn (as it was before without the extra param)
2. hideColumn that ONLY hides the column.
 
So for this specific line of code put it back to "if (model.fields[i].unsupported || model.fields[i].isReadOnly)"